mintpy/prep_fringe.py

```diff
@@ -318,6 +341,105 @@ def prepare_geometry(outfile, geom_dir, box, metadata):
     return outfile
 
 
+def prepare_stack(outfile, unw_file, corr_file, metadata, processor, baseline_dir=None, box=None):
+    print('-'*50)
+    print('preparing ifgramStack file: {}'.format(outfile))
+    # copy metadata to meta
+    meta = {key : value for key, value in metadata.items()}
+
+    # grab date list from the filename
+    unw_files = sorted(glob.glob(unw_file))
+
+    # get the dates from the file names
+    lst_dates12  = []
+    sec_dates    = []
+    for unw_file in unw_files:
+        date12 = os.path.splitext(os.path.basename(unw_file))[0]
+        d12    = date12.split('_')[:2]
+        sec_dates.append(str(d12[1])) # for bperp
+        lst_dates12.append([d12[0].encode('utf-8'), d12[1].encode("utf-8")])
+
+    arr_dates12 = np.array(lst_dates12)
+    ref_date = arr_dates12[0, 0].decode('utf-8')
+    num_pair = arr_dates12.shape[0]
+    print('number of interferograms:', num_pair)
+
+    # get connected component file extension from last ifg
+    cc_file  = f'{os.path.splitext(unw_file)[0]}.conncomp'.replace(
+                                        f'{ref_date}_{sec_dates[-1]}', '*')
+
+    cc_files = sorted(glob.glob(cc_file))
+    if not cc_files:
+        print(f'Could not find any connected component files matching {cc_file}')
+        print('Skipping ifgramStack creation')
+        return
+
+    print('number of associated connected components:', len(cc_files))
+
+    # baseline info
+    if baseline_dir is not None:
+        # read baseline data
+        baseline_dict = isce_utils.read_baseline_timeseries(baseline_dir,
+                                                            processor=processor,
+                                                            ref_date=ref_date)
+        pbase = np.array([baseline_dict[sec][0] for sec in sec_dates],
+                                                            dtype=np.float32)
+
+    # size info
+    if not box:
+        box = (0, 0, int(meta['WIDTH']), int(meta['LENGTH']))
+
+    kwargs = dict(xoff=box[0],
+                  yoff=box[1],
+                  win_xsize=box[2]-box[0],
+                  win_ysize=box[3]-box[1])
+
+    # get correlation
+    ds   = gdal.Open(corr_file, gdal.GA_ReadOnly)
+    corr = np.array(ds.GetRasterBand(1).ReadAsArray(**kwargs), dtype=np.float32)
+    del ds
+
+    # define (and fill out some) dataset structure
+    dropIfgram = np.ones(arr_dates12.shape[0], dtype=bool)
+    ds_name_dict = {
+        "date"       : [np.dtype('S8'), (num_pair, 2), arr_dates12],
+
+        "bperp"      : [np.float32,  (num_pair,), pbase],
+        "dropIfgram" : [np.bool_,    (num_pair,), dropIfgram],
+
+        "coherence"  : [np.float32,  (box[3]-box[1], box[2]-box[0]), corr],
```

Comment:
`coherence` dataset is optional for `ifgramStack`. Since we have only one temporal coherence here, I would recommend not loading coherence at all.